pkg/util/ptr/ptr.go

```diff
@@ -69,3 +69,32 @@ func Overwrite[T any](dst *T, src T) {
 		*dst = src
 	}
 }
+
+// OverwriteWithUser overwrites the *dst with an optional user value depending on
+// if the value needs to be updated based on the current value.
+func OverwriteWithUser[T comparable](dst **T, user, current *T) {
+	if dst == nil {
+		panic("dst is nil")
```

Comment:
can it panic for classless VMs where the dst is not set?.

Reply:
For classless VMs the caller's ConfigSpec will be all default values. And `dst` always needs to point to a field in that  ConfigSpec.